src/reconcile/rec_visibility.c

```diff
@@ -715,6 +806,39 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
     /* Check the update chain for conditions that could prevent it's eviction. */
     WT_RET(__rec_validate_upd_chain(session, r, onpage_upd, select_tw, vpack));
 
+    /* Check if we need to delete any value from the history store. */
```

Comment:
The rec_upd_select function doesn't seem like the best candidate for doing this fixup. It's already one of the most complex functions in the system, and that's partly because we've overloaded it's job of selecting which update should be written to the data file.

Could this history store fixup happen later during the reconciliation process?

Reply:
It can happen later when we inserting the updates to the history store. However, it is also a complex function.

Reply:
Fixed in [ffea256](https://github.com/wiredtiger/wiredtiger/pull/7939/commits/ffea25641f7d62ac1919f643ba0d12ae1b3c2075)